validated/dash_apps/finished_apps/data_report.py

```diff
@@ -0,0 +1,207 @@
+                            # options=Station.objects.order_by("station_code"),
+                            options=[
+                                item.station_code
+                                for item in Station.objects.order_by("station_code")
+                            ],
+                            value=init_station.station_code,
+                        ),
+                        html.H3("Variable"),
+                        dcc.Dropdown(
+                            id="variable_drop",
+                            options=[
+                                {"label": item.name, "value": item.variable_code}
+                                for item in Variable.objects.order_by("variable_code")
+                            ],
+                            value=init_variable.variable_code,
+                        ),
+                        html.H3("Start date - End date"),
+                        dcc.DatePickerRange(
+                            id="date_range_picker",
+                            display_format="YYYY-MM-DD",
+                            start_date=init_start_time,
+                            end_date=init_end_time,
+                        ),
+                    ],
+                ),
+                html.Div(
+                    style={"width": "65%"},
+                    children=[
+                        dcc.Graph(
+                            id="data_report_graph",
+                            figure=plot,
+                        ),
+                    ],
+                ),
+            ],
+        ),
+    ]
+)
+
+
+@app.callback(
+    Output("data_report_graph", "figure"),
+    [
+        Input("temporality_drop", "value"),
+        Input("station_drop", "value"),
+        Input("variable_drop", "value"),
+        Input("date_range_picker", "start_date"),
+        Input("date_range_picker", "end_date"),
+    ],
+)
+def update_graph(
+    temporality: str,
+    station: str,
+    variable: str,
+    start_time: str,
+    end_time: str,
+) -> px.line:
+    station = Station.objects.get(station_code=station)
+    variable = Variable.objects.get(variable_code=variable)
+
+    plot = plot_graph(
+        temporality,
+        station,
+        variable,
+        start_time,
+        end_time,
+    )
+
+    return plot
+
+
+@app.callback(
+    [
+        Output("temporality_drop", "value"),
+        Output("station_drop", "value"),
+        Output("variable_drop", "value"),
+        Output("date_range_picker", "start_date"),
+        Output("date_range_picker", "end_date"),
+    ],
+    Input("clear_button", "n_clicks"),
+    prevent_initial_call=True,
+)
+def clear_form(n_clicks: int):
+    return (
+        init_temporality,
+        init_station.station_code,
+        init_variable.variable_code,
+        init_start_time,
+        init_end_time,
+    )
+
+
+@app.callback(
+    Output("alert_div", "children"),
+    Input("data_report_graph", "figure"),
+)
+def clear_form(figure):
```

Comment:
Just to make sure I get this right, there are two functions called the same, and all works ok?